site-src/geps/gep-1757.md

```diff
@@ -0,0 +1,224 @@
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: NamedAddress
+    value: some-existing-service
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST mark the Gateway as Ready and provide an address in `Status.Addresses` where the Gateway can be reached.
+  This MUST be derived from the referenced `Spec.Address`.
+* MUST not deploy any resources into the cluster; it is expected that a user will do these actions.
+
+When an implementation uses `Service`:
+* MUST have `Spec.Address` be a `NamedAddress` referring to the Service by name. Cross namespace references are not allowed.
+
+Any other "existing infrastructure" types re implementation specific
+
+### Merging Gateways
+
+The spec currently specifies that gateways *can* be merged. This MUST be implemented by referencing the same "existing infrastructure" (as defined in "Manual Deployments").
+
+For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: merged-gateway
+spec:
+  addresses:
+  - type: NamedAddress
+    value: parent-gateway
+  gatewayClassName: example
+  listeners:
+  - name: some-other-route
+    port: 81
+    protocol: HTTP
+```
+
+With this configuration, an implementation:
+* MUST treat the `parent-gateway` as having the concatenated list of all listeners from merged Gateways.
+  This means, for example, that a Service with port 80 and 81 may be created in this example.
+
+### Customizations
+
+With any in-cluster deployment, customization requirements will arise. 
+
+Some common requirements would be:
+* `Service.spec.type`, to control whether a service is a `ClusterIP` or `LoadBalancer`.
+* If it is a `LoadBalancer`, the `Service.spec.loadBalancerIP` to assign to it.
+* Arbitrary labels and annotations on generated resources.
+* CPU and memory requests, and other similar customizations.
+
+For setting a specific IP address in the Service, the `.Spec.Address` field can be used. For example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: IPAddress
+    value: 1.1.1.1
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+This differs from the selection of "manual deployment", which doesn't use `IPAddress` type.
+
+Currently, all other customizations are implementation specific.
+Temporary Note: it seems like most implementations are using `parametersRef` in GatewayClass. The author thinks this is a bit problematic.
+
+### Resource Attachment
+
+Generated resources are guaranteed to have two attributes:
+* A `gateway.networking.k8s.io/metadata.name: <NAME>` label.
```

Comment:
To clarify, this is only for resources generated for a Gateway? If there exists a future use case where resources are generated for some other resource in Gateway API, would we have a way of further distinguishing that? Also have a similar question about if we should try to specify the namespace of the original resource or if that's always going to be the same.

Reply:
Yes, this is for resources generated for the Gateway. IE Service and Deployment.

I was assuming same namespace. I can put that in the spec, although it may be too limiting. So far I am being opinionated until someone suggests otherwise though, so I will make it a MUST for now...